BLSQ/iaso
# POLIO-1655 adapt stock backend

Backend was ignoring usable vials on some incident reports

Related JIRA tickets : POLIO-1655

## Self proofreading checklist


## Doc
NA

## Changes

- LOSSES, STEALING, BROKEN and RETURN incident types now check only usable vials and don't update unusable vials
- Commented out OTHER, since it's still being specced
-  Fixed an (unrelated) failing test for vaccine authorizations
- Update some logic:
    - vials received: only take into account vials from arrival reports. Exclude those found in physical inventory
    - vials used: aggregate vials from form A outgoing movement i.o using the difference between received and usable vials
    - Add methods to VaccineStockCalculator to make these calculations
    - Refactor existing methods (mostly because the new methods just extract some code from the old ones)
- Update and comment tests

## How to test

Go to vaccine stock management
Add one of the listed incident report types to an existing stock
Save
Go back to stock list
The incident report should appear, in usable vials table, with the amount in vials OUT. It should not appear in the unusable vials table

## Print screen / video

https://github.com/user-attachments/assets/c4cb3049-57a1-4fe1-a69a-328f4613e3a7


## Notes

merges into https://github.com/BLSQ/iaso/pull/1575
